src/main/java/de/zalando/aruha/nakadi/metrics/EventTypeMetricRegistry.java

```diff
@@ -0,0 +1,36 @@
+package de.zalando.aruha.nakadi.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventTypeMetricRegistry {
+
+    private final Map<String, EventTypeMetrics> metricsPerEventType = new HashMap<>();
+    private final MetricRegistry metricRegistry;
+
+
+    public EventTypeMetricRegistry(final MetricRegistry metricRegistry) {
+        this.metricRegistry = metricRegistry;
+    }
+
+    public EventTypeMetrics metricsFor(final String eventTypeName) {
+        EventTypeMetrics eventTypeMetrics = metricsPerEventType.get(eventTypeName);
+        if (eventTypeMetrics == null) {
+            synchronized (metricsPerEventType) {
+                eventTypeMetrics = getOrAddMetricsFor(eventTypeName);
+            }
+        }
+        return eventTypeMetrics;
+    }
+
+    private EventTypeMetrics getOrAddMetricsFor(final String eventTypeName) {
+        EventTypeMetrics eventTypeMetrics = metricsPerEventType.get(eventTypeName);
+        if (eventTypeMetrics == null) {
```

Comment:
Mult-threading is hard. One way of not having to handle it ourselves would be to use a guava cache here. Wdyt?